tests/test_pages.py

```diff
@@ -33,6 +34,166 @@ def to_item(self) -> dict:
     }
 
 
+@pytest.mark.asyncio
+async def test_multi_layout_page_object():
+    @attrs.define
+    class Header:
+        text: str
+
+    class H1Page(WebPage[Header]):
+        @field
+        def text(self) -> Optional[str]:
+            return self.css("h1::text").get()
+
+    class H2Page(WebPage[Header]):
+        @field
+        def text(self) -> Optional[str]:
+            return self.css("h2::text").get()
+
+    @attrs.define
+    class HeaderMultiLayoutPage(MultiLayoutPage[Header]):
+        response: HttpResponse
+        h1: H1Page
+        h2: H2Page
+
+        async def get_layout(self) -> ItemPage[Header]:
+            if self.response.css("h1::text"):
+                return self.h1
+            return self.h2
+
+    html_h1 = b"""
+    <!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <title>h1</title>
+        </head>
+        <body>
+            <h1>a</h1>
+        </body>
+    </html>
+    """
+    html_h2 = b"""
+    <!DOCTYPE html>
+    <html lang="en">
+        <head>
+            <title>h2</title>
+        </head>
+        <body>
+            <h2>b</h2>
+        </body>
+    </html>
+    """
+
+    response1 = HttpResponse("https://example.com", body=html_h1)
+    h1_1 = H1Page(response=response1)
+    h2_1 = H2Page(response=response1)
+    response2 = HttpResponse("https://example.com", body=html_h2)
+    h1_2 = H1Page(response=response2)
+    h2_2 = H2Page(response=response2)
+
+    item1 = await HeaderMultiLayoutPage(response=response1, h1=h1_1, h2=h2_1).to_item()
+    item2 = await HeaderMultiLayoutPage(response=response2, h1=h1_2, h2=h2_2).to_item()
+
+    assert item1.text == "a"
+    assert item2.text == "b"
+
+
+@pytest.mark.asyncio
+async def test_multi_layout_page_object_shared_partial_layout():
+    """Scenario where a multi-layout page object acts as a switch for 2 or
+    more layout page objects that all inherit from some other page object class
+    that implements extraction for shared fields."""
+
+    @attrs.define
+    class PartialItem:
+        url: str
+
+    @attrs.define
+    class FullItem(PartialItem):
+        text: str
+
+    class PartialPage(WebPage[PartialItem]):
+        @field
+        async def url(self) -> str:
+            return str(self.response.url)
+
+    class FullPage1(PartialPage, Returns[FullItem]):
+        @field
+        async def text(self) -> Optional[str]:
+            return self.css("h1::text").get()
+
+    class FullPage2(PartialPage, Returns[FullItem]):
+        @field
+        async def text(self) -> Optional[str]:
+            return self.css("h2::text").get()
+
+    @attrs.define
+    class MyMultiLayoutPage(MultiLayoutPage[FullItem]):
+        response: HttpResponse
+        page1: FullPage1
+        page2: FullPage2
+
+        async def get_layout(self) -> ItemPage[FullItem]:
```

Comment:
```suggestion
        async def get_layout(self) -> ItemPage:
```

Removes the need for the `type: ignore[return-value]` below. 

This should be okay since we're expecting `ItemPage` (and its subclasses) to be returned. For some reason, mypy fixates on the `FullItem` return type.